File: sample/src/main/java/com/schibsted/spain/barista/sample/TextListViewAdapter.java
package com.schibsted.spain.barista.sample;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

public class TextListViewAdapter extends BaseAdapter {
  private Activity activity;
  private String[] items;
  private final TextView clickedResult;

  TextListViewAdapter(Activity activity, String[] items, TextView clickedResult) {
    this.activity = activity;
    this.items = items.clone();
    this.clickedResult = clickedResult;
  }

  @Override
  public int getCount() {
    return items.length;
  }

  @Override
  public Object getItem(int position) {
    return items[position];
  }

  @Override
  public long getItemId(int position) {
    return position;
  }

  @Override
  public View getView(final int position, View convertView, ViewGroup parent) {
    View rowView = convertView;

    if (convertView == null) {
      LayoutInflater inflater = (LayoutInflater) activity.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
      rowView = inflater.inflate(R.layout.row_with_buttons, parent, false);
    }

    TextView textView = (TextView) rowView.findViewById(R.id.textview);
    View yesButton = rowView.findViewById(R.id.yes);
    View noButton = rowView.findViewById(R.id.no);
    textView.setText(items[position]);

    rowView.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View view) {
        clickedResult.setText(ListsActivity.getComplexListViewTextAt(position));
      }
    });
    yesButton.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View view) {
        clickedResult.setText("yes");

      }
    });
    noButton.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View view) {
        clickedResult.setText("no");
      }
    });

    return rowView;
  }
}

File: sample/src/main/java/com/schibsted/spain/barista/sample/TextListViewAdapter.kt
package com.schibsted.spain.barista.sample

import android.app.Activity
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

class TextListViewAdapter internal constructor(private val activity: Activity, items: Array<String>, clickedResult: TextView?) :
  BaseAdapter() {
  private val items: Array<String>
  private val clickedResult: TextView?
  override fun getCount(): Int {
    return items.size
  }

  override fun getItem(position: Int): Any {
    return items[position]
  }

  override fun getItemId(position: Int): Long {
    return position.toLong()
  }

  override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
    var rowView = convertView
    if (convertView == null) {
      val inflater = activity.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
      rowView = inflater.inflate(R.layout.row_with_buttons, parent, false)
    }
    val textView = rowView?.findViewById<View>(R.id.textview) as TextView
    val yesButton = rowView?.findViewById<View>(R.id.yes)
    val noButton = rowView?.findViewById<View>(R.id.no)
    textView.text = items[position]
    rowView?.setOnClickListener { clickedResult!!.setText(ListsActivity.Companion.getComplexListViewTextAt(position)) }
    yesButton.setOnClickListener { clickedResult!!.text = "yes" }
    noButton.setOnClickListener { clickedResult!!.text = "no" }
    return rowView
  }

  init {
    this.items = items.clone()
    this.clickedResult = clickedResult
  }
}